### PR TITLE
Update Project to CircleCI 2.0

### PR DESCRIPTION
**Pull request by alexfornuto**
## Effect
PR includes the following changes:
- Updates the documentation project to use CircleCI 2.0
- Uses the Docker image `pantheonsystems/documentation`

## Remaining Work
- [x] Testing
- [x] Testing
- [x] Fix CSS on staging environment
- [x] Test deployment
- [x] Re-implement HTML proofing
- [x] Testing
- [x] Change docker tag back to `latest`
- [x] Change deployment script pack to proper `PROD_UUID`
- [x] Change deployment logic in Circle config to look at `master`.

## Post Launch
To be completed by the docs team upon merge: 
- [ ] Update Status Report
- [ ] Archive from **Done** in Waffle

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - run:
+          name: htmlProofer
+          command: |
+            htmlproofer --assume-extension  ./output_dev/ --disable-external true
+      - store_artifacts:
+          path: ~/build/sitespeed-result
+
+      - deploy:
+          command: |
+            if [ "${CIRCLE_BRANCH}" == "masterblaster" ]; then
```
Comment (stevector) [inline .circleci/config.yml]: This branch name should change before merging.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      - image: pantheonsystems/documentation:circle
+    working_directory: ~/build
+    steps:
+      - checkout
+      - run:
+          name: Generate build
+          command: |
+            npm install
+            ./node_modules/.bin/grunt
+            /documentation/bin/sculpin generate --env=prod --quiet
```
Comment (stevector) [inline .circleci/config.yml]: @alexfornuto why does `sculpin generate` get run in multiple places? Here and `deploy-multidev`/`deploy-live`

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          name: Deploy Multidev environment
+          command: bash ./scripts/deploy-multidev.sh
+      - run:
+          name: Start Ghost Driver
+          command: phantomjs --webdriver=8643
+          background: true
+      - run:
+          name: Start Sculpin
+          command: /documentation/bin/sculpin server
+          background: true
```
Comment (stevector) [inline .circleci/config.yml]: It seems odd that some tests run against a local sculpin site and some run against the remote site. But let's not block the Circle 1 -> Circle 2 migration on that.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+      - deploy:
+          command: |
+            if [ "${CIRCLE_BRANCH}" == "masterblaster" ]; then
+              scripts/deploy-live.sh
+            fi
+
+
+# artifacts need to be done sitespeed-result
+# deployment command for master: bash scripts/deploy-live.sh
```
Comment (stevector) [inline .circleci/config.yml]: This comment looks outdated.

File: .circleci/config.yml (hunk 3 above)
Comment (alexfornuto) [reply .circleci/config.yml]: I believe that the only tests run against the remote environment are the speedtests, where it makes sense to test the speed on the platform.

File: .circleci/config.yml (hunk 2 above)
Comment (alexfornuto) [reply .circleci/config.yml]: It needs to be generated in the multidev script with certain views changed for the multidev environment name. With that in mind, I suppose I'll keep generation there and in live, and test removing it here.

**Comment by alexfornuto** [comment]
holdon

**Pull request merged**